tests/semantic_analyser.cpp

```diff
@@ -78,31 +82,36 @@ void test(BPFtrace &bpftrace,
   // Override to mockbpffeature.
   bpftrace.feature_ = std::make_unique<MockBPFfeature>(mock_has_features);
   ast::SemanticAnalyser semantics(driver.root.get(), bpftrace, out, has_child);
-  ASSERT_EQ(expected_result, semantics.analyse()) << msg.str() + out.str();
+  if (expected_result == -1) {
+    // Accept any failure result
+    EXPECT_NE(0, semantics.analyse()) << msg.str() + out.str();
+  } else {
+    EXPECT_EQ(expected_result, semantics.analyse()) << msg.str() + out.str();
```

Comment:
Similarly to @jordalgo's comment above, why use `EXPECT_EQ` here instead of `ASSERT_EQ`? Same below for testing the error message.

Reply:
I prefer to use EXPECT unless ASSERT is required due to later code assuming success.

In this case, it's safe to check the error message no matter what the return code is, so EXPECT is ok, and it can provide the test developer with more information about why their test is failing.

e.g. `expected_error_code = 10`, `expected_error_message = "undefined map"`
If the actual error code ends up being `1`, then it'd speed up debugging to also show what the actual error message is (e.g. "invalid timestamp mode"). Just means that the developer won't have to go and run the script manually in bpftrace to work out the problem.

Reply:
Makes sense, thanks!